src/archive_open.c

```diff
@@ -20,6 +20,10 @@
 #include <errno.h>
 #include <stdlib.h>
 
+#ifdef __MINGW32__
```

Comment:
What's fcntl.h needed for? Is this for setmode below? If yes, then this should be #ifdef _WIN32.